tmt/steps/report/html.py

```diff
@@ -44,19 +49,20 @@ class ReportHtml(tmt.steps.report.ReportPlugin):
     _keys = ["open"]
 
     @classmethod
-    def options(cls, how=None):
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
         """ Prepare command line options for the html report """
-        return [
-            click.option(
-                '-o', '--open', is_flag=True,
-                help='Open results in your preferred web browser.'),
-            ] + super().options(how)
+        options = super().options(how)
+        options.append(click.option(
+            '-o', '--open', is_flag=True,
+            help='Open results in your preferred web browser.'))
+        return options
```

Comment:
@psss PTAL. I've changed the code so it doesn't require `cast(List[ClickOptionDecoratorType]]`. The resulting list, however, contains elements in a different order than before (the newly created option is now last instead of first). Is that an issue?

Reply:
This could also get a short explanation why am I doing it this way. It was briefly discussed on irc but I'm writing it here so it doesn't get lost easily.

In the original code, we were joining two lists which should be type `List[ClickOptionDecoratorType]`:
```
return [click.option()] + super().options()
```
However, `mypy` cannot automatically deduce the type of the first list `[click.option()]` to be a compatible type with `List[ClickOptionDecoratorType]`. These lists are compatible, `mypy` just cannot deduce it. Thus, the first option that we used was to cast the first list.

But it can be improved so we can avoid casting by annotating the list before assignment:
```
options: List[tmt.options.ClickOptionDecoratorType] = [click.option()]
return options + super().options()
```
This variant preserves the order of elements in the list, if it's important to do so.

However, it can be even shortened so we don't need to annotate the variable:
```
options = super().options()
options.append(click.option())
return options
```
Since the method `super().options()` is properly typed to return `List[ClickOptionDecoratorType]`, `mypy` can then clearly deduce that the `options` variable will be the same type. And appending the `click.option()` is a compatible operation. Note that the operation was incompatible when we used 2 lists instead of 1.

Reply:
Thanks much for the detailed explanation. It would be good to preserve the order of the options so that the plugin-specific ones are listed in the `--help` message at the top. But we can achieve that simply by using list slice. Adjusted.